editor/src/components/canvas/canvas-strategies/strategies/group-conversion-helpers.ts

```diff
@@ -773,8 +785,42 @@ export function createWrapInGroupActions(
 
   const groupPath = EP.appendToPath(parentPath.intendedParentPath, group.uid) // TODO does this work if the parentPath is a ConditionalClauseInsertionPath?
 
-  const pinChangeCommands = childComponents.flatMap((childComponent) => {
-    const newChildPath = EP.appendToPath(groupPath, childComponent.element.uid)
+  const retargetedChildComponentsForPinChanges: Array<{
+    element: JSXElement
+    metadata: ElementInstanceMetadata
+    expectedElementPath: ElementPath
+  }> = orderedActionTargets.flatMap((p) => {
+    return replaceFragmentLikePathsWithTheirChildrenRecursive(
+      metadata,
+      allElementProps,
+      elementPathTrees,
+      [p],
+    ).flatMap((maybeRetargetedPath) => {
```

Comment:
I think this `flatMap` should just be a regular `map`, I'm also a little wary about the number of `flatMap` calls in here as well if it's on the performance hot path.

Reply:
fixed!